app/contributing/kong-plugins.md

```diff
@@ -0,0 +1,120 @@
+            versions.yml
+    ```
+
+1. Populate the files in the directory with your own info:
+
+    * `_metadata.yml`: Sets the metadata for the plugin. 
+    Follow the instructions in the file to fill it out.
+
+    * `_index.md`: Introduction for your plugin. This is where you
+    explain how the plugin works and why someone would want to use it.
+
+    * `_changelog.md`: A changelog for your plugin. For the first entry,
+    just note when the plugin was published. 
+
+    * `how-to/_index.md`: Markdown documentation on how to use the plugin.
+      You can create any number of files in the `how-to` folder, e.g. 
+      `how-to/_getting-started.md`, `how-to/_metrics.md`, etc.
+
+    * `versions.yml`: Set the {{site.base_gateway}} version that the plugin is introduced in.
+        This will generate a doc for every subsequent gateway version, 
+        starting with the one you specify.
+
+1. Add an icon for your plugin into the 
+[`/app/_assets/images/icons/hub`](https://github.com/Kong/docs.konghq.com/tree/main/app/_assets/images/icons/hub) 
+directory. 
+    
+    Plugin icons are required for publication on the Kong plugin hub. Icons
+    should be a PNG or SVG file, 120x120 pixels in size. 
+
+    The filename of your image should be `kong-inc_plugin-name`.
+    For example, `kong-inc_oas-validation`.
+
+1. Create a basic example for your plugin in the [plugin toolkit repository](https://github.com/fabianrbz/kong-plugins-docs-toolkit/tree/main/examples).
+
+    This example will be validated against the plugin's schema,
+    so make sure to include all required configuration parameters.
+
+## Adding images
+
+If you have any diagrams or screenshots that you want to add to your plugin documentation:
+
+1. Add to the images into the `app/_assets/images/docs/plugins` directory.
+
+   Make sure that any screenshots follow the [screenshot guidelines](/contributing/user-interfaces/#screenshots).
+
+1. Insert images into any of the markdown files for your plugin using the following format:
+
+    ```
+    ![Authentication flow diagram](/assets/images/docs/plugins/my-plugin-auth-flow.png)
+    > *Figure 1: Diagram showing an OAuth2 authentication flow with Keycloak.*
+    ```
+
+## Test and submit plugin
+
+1. Run the docs site locally per the instructions in
+the docs [README](https://github.com/Kong/docs.konghq.com#run-locally).
+
+   You should find your Hub contribution listed at `localhost:3000/hub`.
+
+1. Once you are happy with your listing, push your branch to the GitHub repository:
+
+    ```
+    git push --set-upstream origin [name_of_your_new_branch]
+    ```
+
+1. Make a pull request against the [docs.konghq.com](https://github.com/Kong/docs.konghq.com/) 
+repository to add your documentation to the Plugin Hub. 
+
+The Kong docs team will review your PR, suggest improvements and adjustments as
+necessary, and once approved, will merge and deploy your Plugin Hub addition!
+
+## Custom plugins and documentation
```

Comment:
Either one. You might be an SE that has a custom plugin, or wants to make one. This is basically for the Kong audience, with callouts to external contributors going "hey, you're in the wrong place". 

It might be a good idea to also add some of this info to the partner plugin page.